melissa/profile_populator.py

```diff
@@ -89,9 +90,32 @@ def empty(variable):
     if empty(city_name):
         city_name = 'New Delhi'
 
-    city_code = raw_input('Code of city from weather.com: ')
+    city_code = raw_input('Code of city from weather.com\
+ or <ENTER> for a search on based on the name\
+ of the city you live in: ')
     if empty(city_code):
-        city_code = 'INXX0096'
+        city_list = pywapi.get_loc_id_from_weather_com(city_name)
+        if city_list['count'] == 0:
+            print 'Sorry, search results were empty.'
+            city_code = 'INXX0096'
+        else:
+            print 'Cities returned are: '
+            for city_list_i in range(city_list['count']):
+                print str(city_list_i+1) + ": " + city_list[city_list_i][1]
+            while(True):
+                city_choice_i = ''
+                city_choice_i = raw_input('Enter the index of \
+the city of your choice: ')
+                if empty(city_choice_i):
+                    city_code = 'INXX0096'
+                    break
+                city_choice_i = int(city_choice_i)
+                if 1 <= city_choice_i <= city_list['count']:
+                    city_code = city_list[city_choice_i-1][0]
+                    break
+                else:
+                    print 'Enter an index from one of the choices,\
+ Try Again!'
```

Comment:
I made the comma a full stop and made Again -> again.
So now it shouldn't be grammatically wrong.